### PR TITLE
Wildcard V1: `<Button />`

### PR DESCRIPTION
**Pull request by umpox**
`import { Button } from '@sourcegraph/wildcard'`

Adds `<Button />` wrappers.

## Button
Simple wrapper around our existing `<button>` usage:

It supports:
- Bootstrap styles through `variant` props
- Different sizes currently supported through Bootstrap
- Rendering as different elements (e.g. `<a>` tag) where required

Notes:
- Migration ticket: https://github.com/sourcegraph/sourcegraph/issues/23220
- Demo: https://5f0f381c0e50750022dc6bf7-gvoaphyqfl.chromatic.com/?path=/story/wildcard-buttons-button--simple

Closes https://github.com/sourcegraph/sourcegraph/issues/22841

### CODE REVIEW

File: client/wildcard/src/components/Buttons/LoadingButton/LoadingButton.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * A wrapper around the `<Button />` component that allows consumers to render a loading spinner consistently.
+ *
+ * This component should typically be used for asynchronous actions (e.g. fetching data from a remote server).
+ *
+ * Be mindful of how the output of the action is signalled for the user.
+ * For example, it may be appropriate to use the 'success' variant once the action has completed successfully.
+ * Likewise, you may want to use the 'danger' variant if the action failed.
+ */
+export const LoadingButton: React.FunctionComponent<LoadingButtonProps> = ({
```
Comment (5h1rU) [inline client/wildcard/src/components/Buttons/LoadingButton/LoadingButton.tsx:31]: What do you think about omitting this `<LoadingButton/>` as a component and instead use a prop `loading` on `<Button loading={true}>`?

Comment (umpox) [reply client/wildcard/src/components/Buttons/LoadingButton/LoadingButton.tsx:31]: I thought about that, mentioned a bit about in on the description above. This is something which I'm keen to get thoughts on.

The reason I didn't add this to `<Button />` is it essentially means adding two new props `loading` and `alwaysShowChildrenWhenLoading`. I'm wary of letting our components become too 'configurable' with magic props. That being said, maybe this is a good exception to make for `loading`

Other option would be to remove `<LoadingButton />` and create something like `<InlineLoadingSpinner />` then consumers can just set that as children when they want to show a loading state.

WDYT?

**Comment by umpox** [comment]
Open discussions:
1. These components currently do not export any styles, they just assume from our Bootstrap classes that are set in the web app. Any preferences on what we should do here? We still want to support the global Bootstrap styles until we fully migrate all buttons
2. `<LoadingButton />` or `<Button loading={true} />` or <Button>{loading && <LoadingSpinnerThing />}</Button>`? Discussion: https://github.com/sourcegraph/sourcegraph/pull/23167#discussion_r676776322

File: client/wildcard/src/components/Buttons/LoadingButton/LoadingButton.tsx (hunk 1 above)
Comment (5h1rU) [reply client/wildcard/src/components/Buttons/LoadingButton/LoadingButton.tsx:31]: Why do you need `alwaysShowChildrenWhenLoading` if you already have a `loading` prop?


More flexible, the spinner can be positioned as before/after, would work without text.
```js
<Button>{loading ? <><IconSpinner/>Saving...</> : "Save"}</Button>
```

Opinionated only one way to show the spinner
```js
<Button loading={loading}>{loading ? 'Saving...' : 'Save'}</Button>
```

It depends on how opinionated we want  buttons components any of the signatures above are good to me

Comment (umpox) [reply client/wildcard/src/components/Buttons/LoadingButton/LoadingButton.tsx:31]: It's trying to provide a similar interface to our current `<LoaderButton />`, that supports rendering the spinner with or without its children in many different places.

After thinking more about this, I do think it's just easier to treat it as `children`, and will be more future-proof. Will refactor

**Review by 5h1rU — APPROVED** [review]
💪

**Pull request merged**